### PR TITLE
Dealing with symlink pointing to directories (fix #11)

### LINKED ISSUE #11: BUG: shell-safe-rm can't deal with symbolic links on Linux
`shell-safe-rm` can't deal with symbolic links pointing to directories, for example:
```
$ export SAFE_RM_DEBUG=1
$ cd /tmp/
$ mkdir test
$ ln -s test test_link
$ rm test_link
[D] file test_link
[D] 1 files or directory to process: test_link
[D] ls_result: test_link
[D] 216: test_link: is a directory
safe-rm: test_link: is a directory
[D] remove returned status: 1
[D] EXIT_CODE 1
```
The only way to delete a link to a directory is to pass `-r` parameter, unlike the standard `rm` will do:
```
$ rm -r test_link
[D] short option -r
[D] file test_link
[D] recursive    : -r
[D] 1 files or directory to process: test_link
[D] ls_result: test_link
[D] trash test_link
[D] mv test_link to /home/developer/.local/share/Trash/files/test_link-2017-03-31_12:25:11-0
[D] 254: trash returned status 0
[D] remove returned status: 0
[D] EXIT_CODE 0
```
I think I've resolved this issue, I will submit a PR

### PR DESCRIPTION
**Pull request by bunop**
Now shell-safe-rm will consider a symlink to a directory like a regular file; Added indentation in `directory` condition

### CODE REVIEW

File: bin/rm.sh [hunk 1]
```diff
@@ -208,51 +208,51 @@ fi
...
-  # if a directory, and without '-r' option
-  if [[ ! -n "$OPT_RECURSIVE" ]]; then
-    debug "$LINENO: $file: is a directory"
-    echo "$COMMAND: $file: is a directory"
-    return 1
-  fi
+  # if is dir and not a symbolick link
+  if [[ -d "$file" && ! -L "$file" ]]; then
+
+    # if a directory, and without '-r' option
```
Comment (bunop) [inline bin/rm.sh]: Added one level of indentation: those are clauses evaluated in `directory` condition

File: Makefile [hunk 2]
```diff
@@ -13,5 +13,18 @@ uninstall:
 	@echo "and do 'unalias rm' from all your terminal sessions"
 	@echo "Successfully removed $(PREFIX)/safe-rm"
 
+# the test target will not work properly if a file named test is ever created
+# in this directory. Since it has no prerequisites, test would always be considered
+# up to date and its recipe would not be executed. To avoid this problem you
+# can explicitly declare the target to be phony by making it a prerequisite of
+# the special target
+# TODO: is test directory really needed?
```
Comment (bunop) [inline Makefile]: I see that there is a `test`directory, but I didn't understand if such code is useful or not. I need to define the `.PHONY` [target rule](http://www.gnu.org/savannah-checkouts/gnu/make/manual/html_node/Phony-Targets.html) in order to do tests

**Comment by bunop** [comment]
This will solve #11

**Comment by kaelzhang** [comment]
temporarily merge

**Pull request closed without merging**